### PR TITLE
refactored loading transaction json tests

### PR DESCRIPTION
**Pull request by debris**
still todo (in next prs):
- completely get rid of `FromJson` trait
- move uint and hash deserialization to rpc module
- make uint deserialization lenient (so it accepts, plain numbers)

### CODE REVIEW

File: ethcore/src/json_tests/transaction.rs [hunk 1]
```diff
@@ -16,75 +16,54 @@
...
-		let rlp = Bytes::from_json(&test["rlp"]);
-		let res = UntrustedRlp::new(&rlp).as_val().map_err(From::from).and_then(|t: SignedTransaction| t.validate(schedule, schedule.have_delegate_call));
-		fail_unless(test.find("transaction").is_none() == res.is_err());
-		if let (Some(&Json::Object(ref tx)), Some(&Json::String(ref expect_sender))) = (test.find("transaction"), test.find("sender")) {
+		let mut fail_unless = |cond: bool| if !cond && !fail { failed.push(name.clone()); println!("Transaction failed: {:?}", name); fail = true };
+
+		let number: Option<u64> = test.block_number.map(Into::into);
+		let schedule = match number {
+			None => &old_schedule,
+			Some(x) if x < 1_000_000 => &old_schedule,
```
Comment (gavofyork) [inline ethcore/src/json_tests/transaction.rs]: should be `1_150_000`

**Comment by gavofyork** [comment]
does this whole way of deserialisation now mean that we do an extra set of copies? i.e. JSON -> ethjson::struct, then ethjson::struct -> eth::struct ? that seems rather wasteful if so.

**Comment by debris** [comment]
> does this whole way of deserialisation now mean that we do an extra set of copies?

no values are copied. There are just move operations.

**Comment by gavofyork** [comment]
looks ok to me...

**Pull request merged**